src/hook.rs

```diff
@@ -165,6 +166,14 @@ pub fn apply_env(
             }
             output::print_activation_to_tty(activation, shadowenv.features());
         }
+        VariableOutputMode::XonshMode => {
+            for (k, v) in shadowenv.exports()? {
+                match v {
+                    Some(s) => println!(r#"${} = "{}""#, k, s.replace(r#"""#, r#"\""#)),
```

Comment:
Is shellescape as used in the other shells truly not sufficient for xonsh?

Reply:
`shell_escape` adds single quotes surrounding the string when there are characters to escape, and does not do this when there are no characters to escape.

Xonsh is expecting a valid Python statement when setting environment variables e.g.: `$MY_VAR = "a string"` so when there aren't any valid characters to escape, we have to add quotation marks after calling `shell_escape` or the statement will be invalid. e.g.: `$MY_VAR = unquoted_value_interpreted_as_variable`

This could be handled with a ternary or conditional statement that adds the quotation marks if they aren't present. But the other problem is that `shell_escape` will escape characters that don't need to be escaped for Xonsh, since in this case all environment variables are Python strings.

Therefore I thought it would be simplest to just default to adding quotation marks around the variable value and escape any double quotes that occurred within the variable value.

Reply:
3. Use [`escape_default`](https://doc.rust-lang.org/std/primitive.char.html#method.escape_default)

This seems to be the most straightforward to me. It's option 2 without having to roll it ourselves. I'm going to commit this and solicit feedback of that solution.

Reply:
Apologies I missed your reply before posting that last response.
> ... it seems like xonsh is pretty esoteric, and too niche to explicitly support.

This is definitely true and a fair argument. 

Just be clear are you suggesting that the init script be something provided outside of shadowenv? Which would mean closing this pull request?